packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -201,6 +236,16 @@ function getTextualContent(element: Element | HTMLElement, userProgrammaticAttri
       removeTextFromElements(`[${userProgrammaticAttribute}]`)
     }
 
+    if (privacyEnabledActionName) {
+      // remove the text of elements with privacy override
+      removeTextFromElements(
+        `[${PRIVACY_ATTR_NAME}=${NodePrivacyLevel.MASK}],
+        [${PRIVACY_ATTR_NAME}=${NodePrivacyLevel.HIDDEN}],
+        [class=${PRIVACY_CLASS_MASK}],
+        [class=${PRIVACY_CLASS_HIDDEN}]`
+      )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think we could extract the logic of having to check both the class and the attribute. Something like:
```ts
  removeTextFromElements(
    getPrivacySelector(NodePrivacyLevel.MASK),
    getPrivacySelector(NodePrivacyLevel.HIDDEN)
  )
```

It could even be used in `getNodeSelfPrivacyLevel` 
```ts
// instead of 
const privAttr = node.getAttribute(PRIVACY_ATTR_NAME)
if (privAttr === PRIVACY_ATTR_VALUE_MASK || elementClasslistContains(node, PRIVACY_CLASS_MASK))

// we could use 
if(node.matches(getPrivacySelector(NodePrivacyLevel.MASK))
```
@BenoitZugmeyer do you see any downside of using [matches](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches), maybe the perf?

Reply:
I think there is a problem to use `.matches()` function. It does not select with desired precedence. For example, `"dd-privacy-mask"` does not select `class="dd-privacy-mask-user-input dd-privacy-mask"`
<img width="584" alt="Screenshot 2024-06-03 at 15 05 43" src="https://github.com/DataDog/browser-sdk/assets/9922567/851fb382-f270-47e6-8eb0-4cb3a87c28c3">

Reply:
I think you forgot the "." in the selector `node.matches(".dd-privacy-mack-user-input")`

Reply:
Ahh sorry I meant `[class="dd-privacy-mask-user-input"]`. I think we could use matches to replace `(privAttr === PRIVACY_ATTR_VALUE_MASK || elementClasslistContains(node, PRIVACY_CLASS_MASK))`, but I don't see how we share the same function with removeTextFromElements.

Reply:
I see. I updated the conditions with this and removed the unused constants and the polyfill function.